### ISSUE TITLE
A few questions about the value net

### ISSUE THREAD

**Issue by isty2e** [gcp/leela-zero#331]
Black was dominating until the recent bug fix, so our training set is filled mostly with games in which black has won, and this messed up the value net quite a bit. The latest network estimates black's winrate on the empty board as something like 88%. A few questions regarding this:

1. Even if the training window is filled with similar number of games won by black/white, since we are not training the network to the convergence, it will take some time for the value net to recover. And we know that as MCTS progresses the value head contributes more than the policy head does, so this might slow down the improvement overall. Would it be desirable to fix it in an artificial way?

2. If so, we can truncate the training set for the moment to exclude the games played before the bug fix. And then, we can probably adjust the loss function for a few more networks from current policy_loss + mse_loss to policy_loss + c * mse_loss (where c > 1), or even train the value head only while fixing the weights in the residual blocks and the policy head. Will it work?

3. Do the suggested methods in 2. violate the _tabula rasa_ principle? We might be introducing human knowledge such that the games should be approximately even, but on the other hand, we are not introducing any domain knowledge or whatsoever, but trying to accomplish the desirable result by just training more (roughly speaking).

**Comment by sethtroisi** [comment]
Some extra debug information:

```
weightsInitEval() { stat -c "%y %n" $1; echo "genmove black" | ./src/leelaz -p 1 --noponder -w $1 |& grep 'NN eval'; }
export -f weightsInitEval
ls autogtp/[1-9a-f][1-9a-f]* -t | xargs -I {} bash -c "weightsInitEval {}"
```

| network | date | NN eval= |
|---|---|---|
|eebb910d | 2017-12-09 | NN eval=0.877569|
|2184b75|2017-12-07 |NN eval=0.923566|
|ad8a1d0|2017-12-06 |NN eval=0.867830|
|6f274ab1|2017-12-03 |NN eval=0.843738|
|22373747|2017-11-30 |NN eval=0.650792|
|92c658d7|2017-11-25 |NN eval=0.492954 |
|a1151640|017-11-24 |NN eval=0.400048|
|1ac2638d|2017-11-21 |NN eval=0.409520|
|857b975e|2017-11-20 |NN eval=0.409467|
|4d9a5cfc|2017-11-19 |NN eval=0.292002|
|fe3f6afd| 2017-11-16 |NN eval=0.309792|

**Comment by kimitsu** [comment]
I wonder if that really contributes to the strength so much? If so, wouldn't it affect one side more than the other? Yet we get a clean ~50% win rate for each color even with the most extreme value estimation of the 2184b75 network. Is there a point to fix something that is, apparently, working anyway? Current network evaluates empty board as 0.825852 for black, so it gradually falls back to normal, do we need to intervene?

**Comment by sethtroisi** [comment]
@kimitsu yep, seems like network is going to unlearn this very quickly.
if ~20k games moved initial eval 5% it might only take ~120k (~4 days) till we're close to 50-50

**Comment by isty2e** [comment]
So my concern is that we need another 200k games to get a better training set, which means for next 200 games we will get some inferior networks, which means that this effect might last longer, like for 500k games or so.

**Comment by evanroberts85** [comment]
It does not seem to affect play a great deal, the relative values between moves are what is important for the MCTS to work well.

**Comment by isty2e** [comment]
That argument assumes that the NN outputs are shifted by a constant value from normal ones, but I wonder if that is the case. Also, an overly optimistic value net will lead to much less risky plays.

**Comment by Yakago** [comment]
I guess this is just another one of those, "We don't really know what's going on" things. 

I.e. we have no evidence that there's a problem, and things are progressing, so there's not much we can do

**Comment by isty2e** [comment]
Closing this issue, since recent networks seems balanced quite a bit.
